### DISCUSSION TITLE
Is there any way to generate C++ bindings with name matching to ICU4C?

### DISCUSSION THREAD

**Discussion by riajain0412** [unicode-org/icu4x#3097] [General]
I want to try out ICU4X but do not want to change name of API's(ICU4C API's) present in my project. Diplomat tool generated C++ bindings, but there name is not same as of ICU4C so is there any way to generate bindings whose name will be similar to ICU4C API's name?

**Comment by Manishearth** [comment]
No, the APIs are designed differently and are not easy to just swap out.

We do plan to eventually make a compatability layer that can sit on top of both ICU4X and ICU4C but that work is just starting, and that layer would also have a different API.

**Comment by sffc** [comment]
@riajain0412 Are you using ICU4C C or C++ APIs (e.g., unum.h or numfmt.h)? And do you have a list?
